api/api/controllers/search_controller.py

```diff
@@ -428,36 +478,78 @@ def search(
-
-    # Paginate
-    start, end = _get_query_slice(s, page_size, page, filter_dead)
-    s = s[start:end]
-    try:
-        if settings.VERBOSE_ES_RESPONSE:
-            log.info(pprint.pprint(s.to_dict()))
+    return s
 
-        search_response = s.execute()
 
-        if settings.VERBOSE_ES_RESPONSE:
-            log.info(pprint.pprint(search_response.to_dict()))
-    except (RequestError, NotFoundError) as e:
-        raise ValueError(e)
+def build_collection_query(
+    s: Search,
+    search_params: media_serializers.MediaListRequestSerializer,
+):
+    """
+    Build the query to retrieve all the items in a collection.
+    The common filters, e.g. license or category, are also applied.
+    :param s: the search client to build the query for.
+    :param search_params: the validated search parameters.
+    :return: the search client with the query applied.
+    """
+    search_query = {"filter": [], "must": [], "should": [], "must_not": []}
+    # Apply the term filters. Each tuple pairs a filter's parameter name in the API
+    # with its corresponding field in Elasticsearch. "None" means that the
+    # names are identical. The `allow_multiple` flag indicates whether the
+    # filter accepts multiple comma-separated values.
+    filters = [
+        # Collection filters allow a single value.
+        ("tag", "tags.name.keyword", False),
+        ("source", "source.keyword", False),
+        ("creator", "creator.keyword", False),
+        # The following filters allow multiple comma-separated values.
+        ("license", "license.keyword", True),
+        ("license_type", "license.keyword", True),
+        ("extension", None, True),
+        ("category", None, True),
+        ("categories", "category", True),
+        ("length", None, True),
+        ("aspect_ratio", None, True),
+        ("size", None, True),
+    ]
+    for serializer_field, es_field, allow_multiple in filters:
+        if serializer_field in search_params.data:
+            arguments = search_params.data.get(serializer_field)
+            if not arguments:
+                continue
+            arguments = arguments.split(",") if allow_multiple else [arguments]
+            parameter = es_field or serializer_field
+            search_query["filter"].append({"terms": {parameter: arguments}})
 
-    results = _post_process_results(
-        s, start, end, page_size, search_response, filter_dead
+    # Exclude mature content and disabled sources
+    include_sensitive_results = search_params.validated_data.get(
+        "include_sensitive_results", False
     )
+    if not include_sensitive_results:
+        search_query["must_not"].append({"term": {"mature": True}})
+    if excluded_providers := get_excluded_providers():
+        search_query["must_not"].append({"terms": {"provider": excluded_providers}})
+
+    # Rank the `tag` results by popularity and authority
+    if "tag" in search_params.data:
```

Comment:
Does this result in the popularity/authority boosting being applied if you have a tag query on a source or creator collection? E.g.: http://localhost:50280/v1/audio/source/jamendo/?tag=peaceful

If so, I think that's potentially confusing behavior since the creator/source endpoints are described as being sorted by ingestion order.